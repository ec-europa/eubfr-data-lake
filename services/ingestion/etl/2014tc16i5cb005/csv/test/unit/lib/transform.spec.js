/**
 * @jest-environment node
 */

import mapper from '../../../src/lib/transform';
import testRecord from '../../stubs/record.json';

describe('2014tc16i5cb005 CSV transformer', () => {
  let result = {};

  beforeAll(() => {
    result = mapper(testRecord);
  });

  test('Returns null when record is not provided', () => {
    expect(mapper()).toBe(null);
  });

  test('Produces correct JSON output structure', () => {
    expect(result).toMatchSnapshot();
  });

  test('Handles varying values for field `Project EU co-financing rate`', () => {
    const copy1 = JSON.parse(JSON.stringify(testRecord));
    const copy2 = JSON.parse(JSON.stringify(testRecord));

    copy1['Project EU co-financing rate'] = '99.00%';
    const copyResult1 = mapper(copy1);
    // In this case percentage is calculated from total, i.e.
    // expect(testRecord['Project final contract amount']).toBe('591955.11');
    // Values less than 100 should be treated as percentages.
    expect(copyResult1.budget.eu_contrib.value).toBe(586035.5589);

    copy2['Project EU co-financing rate'] = '500665.00%';
    const copyResult2 = mapper(copy2);
    // Values greater than 100 should not be treated as percentages.
    expect(copyResult2.budget.eu_contrib.value).toBe(500665.00000000006);
  });
});
