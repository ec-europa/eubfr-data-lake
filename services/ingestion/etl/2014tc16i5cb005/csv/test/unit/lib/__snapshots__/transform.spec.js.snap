// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2014tc16i5cb005 CSV transformer Produces correct JSON output structure 1`] = `
Object {
  "action": "",
  "budget": Object {
    "eu_contrib": Object {
      "currency": "EUR",
      "raw": "85.00%",
      "value": 503161,
    },
    "funding_area": Array [],
    "mmf_heading": "",
    "other_contrib": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "private_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "public_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "total_cost": Object {
      "currency": "EUR",
      "raw": "591955.11",
      "value": 591955,
    },
  },
  "call_year": "",
  "complete": false,
  "description": "Project call: CB005.1 
Project intervention field: 087 Adaptation to climate change measures and prevention and management of climate related risks e.g. erosion, fires, flooding, storms and drought, including awareness raising, civil protection and disaster management systems and infrastructures 
Last modified: 01.12.2017 
",
  "ec_priorities": Array [],
  "media": Array [],
  "programme_name": "",
  "project_id": "CB005.1.11.005",
  "project_locations": Array [
    Object {
      "address": "",
      "centroid": null,
      "country_code": "BG",
      "location": null,
      "nuts": Array [],
      "postal_code": "",
      "region": "Haskovo",
      "town": "",
    },
  ],
  "project_website": "",
  "related_links": Array [],
  "reporting_organisation": "Member states",
  "results": Object {
    "available": "",
    "result": "",
  },
  "status": "",
  "sub_programme_name": "",
  "success_story": "",
  "themes": Array [],
  "third_parties": Array [
    Object {
      "address": "",
      "country": "Bulgaria",
      "email": "",
      "name": "Directorate General Fire Safety and Civil Protection - Ministry of interior",
      "phone": "",
      "region": "Haskovo",
      "role": "coordinator",
      "type": "",
      "website": "",
    },
  ],
  "timeframe": Object {
    "from": "2017-03-21T00:00:00.000Z",
    "from_precision": "day",
    "to": "2019-03-20T00:00:00.000Z",
    "to_precision": "day",
  },
  "title": "Increasing the operational capacity for forest fires and improving prevention of disasters",
  "type": Array [],
}
`;
